harness/determined/cli/trial.py

```diff
@@ -121,6 +126,69 @@ def trial_logs(args: Namespace) -> None:
     )
 
 
+@authentication.required
+def generate_support_bundle(args: Namespace) -> None:
+    output_dir = args.output_dir
+    if output_dir is None:
+        output_dir = os.getcwd()
+
+    elif not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+
+    dt = datetime.now().strftime("%Y%m%dT%H%M%S")
+    tar_filename = f"det-bundle-trial-{args.trial_id}-{dt}.tar.gz"
+    fullpath = os.path.join(output_dir, tar_filename)
+
+    with tempfile.TemporaryDirectory() as temp_dir, tarfile.open(fullpath, "w:gz") as bundle:
+        trial_logs_filepath = write_trial_logs(args, temp_dir)
+        master_logs_filepath = write_master_logs(args, temp_dir)
+        api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)
+
+        bundle.add(trial_logs_filepath, arcname=os.path.basename(trial_logs_filepath))
+        bundle.add(master_logs_filepath, arcname=os.path.basename(master_logs_filepath))
+        bundle.add(api_trail_filepath, arcname=os.path.basename(api_trail_filepath))
+        bundle.add(api_experiment_filepath, arcname=os.path.basename(api_experiment_filepath))
+
+        print(f"bundle path: {fullpath}")
+
+
+def write_trial_logs(args: Namespace, temp_dir: str) -> str:
+    trial_logs = api.trial_logs(args.master, args.trial_id)
+    file_path = os.path.join(temp_dir, "trial_logs.txt")
+    with open(file_path, "w") as f:
+        for log in trial_logs:
+            f.write(log["message"])
+
+    return file_path
+
+
+def write_master_logs(args: Namespace, temp_dir: str) -> str:
+    response = api.get(args.master, "logs")
+    file_path = os.path.join(temp_dir, "master_logs.txt")
+    with open(file_path, "w") as f:
+        for log in response.json():
+            f.write("{} [{}]: {}".format(log["time"], log["level"], log["message"]))
+    return file_path
+
+
+def write_api_call(args: Namespace, temp_dir: str) -> Tuple[str, str]:
+    file_path1 = os.path.join(temp_dir, "api_experiment_call.json")
+    file_path2 = os.path.join(temp_dir, "api_trial_call.json")
+
+    trial_obj = bindings.get_GetTrial(setup_session(args), trialId=args.trial_id).trial
+    experiment_id = trial_obj.experimentId
+    exp_obj = bindings.get_GetExperiment(setup_session(args), experimentId=experiment_id)
+
+    create_json_file_in_dir(exp_obj.to_json(), file_path1)
```

Comment:
`file_path1` and `file_path2` don't really tell me anything, I have to read what they are and how they are used anyway. which is to say maybe just put them where they're used, or closer to where they're used, or give them better names.
```suggestion
    create_json_file_in_dir(exp_obj.to_json(), os.path.join(temp_dir, "api_experiment_call.json"))
```